File: 課題3おまけ/bogo.java
import java.util.Random;

class bogo
{
	public static void main(String args[])
	{
		int size = 1;
		Random rnd = new Random();
		int a[], b[];
		a = new int [1000000];
		b = new int [1000000];

		for (int t = 0; t < 5; t++) {
			size *= 2;
			for (int i = 0; i < size; i++) {
				a[i] = rnd.nextInt(1000);
			}
			long tstart = System.nanoTime();
			while(true) {
				int k = 0;
				int check = 1;
				for (int i = 0; i < size - 1; i++) {
					if (a[i] > a[i + 1]) {
						check = 0;
						break;
					}
				}
				if (check == 1) break;
				for (int j = size; j > 1; j--) {
					int get = rnd.nextInt(j);
					b[k] = a[get];
					k++;
					a[get] = a[j - 1];
				}
				b[k] = a[0];
				for (int i = 0; i < size; i++) {
					a[i] = b[i];
				}
			}
			long tend = System.nanoTime();

			System.out.println((tend - tstart) / 1000000000f);
		}
	}
}

File: 課題3おまけ/bogo.rb
def randoms(id, size, max)
	a = Array.new(size)
	for i in 0.. size - 1
		if max == 1
			a[i]=rand()
		else
			a[i]=rand(max)
		end
	end
	a
end

def main(n)
	size = 1
	for t in 0 .. 3
		size *= 2
		a = randoms(0, size, 999)
		tstart = Time.now
		bogo(a, size)
		tend = Time.now

		puts "#{tend - tstart}"
	end
end

def bogo(a, size)
	b = Array.new(size, 0)
	while true
		k = 0
		check = 1
		for i in 0 .. size - 2
			if a[i] > a[i + 1]
				check = 0
				break
			end
		end
		if check == 1
			break
		end
		for j in 0 .. size - 1 
			i = size - 1
			get = rand(200)
			b[k] = a[get]
			k++
			a[get] = a[i - 1]
		end
		b[k] = a[0]
		for i in 0 .. size - 1
			a[i] = b[i]
		end
	end
end
